x/mongo/driver/topology/connection.go

```diff
@@ -108,7 +109,24 @@ func (c *connection) connect(ctx context.Context) {
 	}
 	defer close(c.connectDone)
 
+	var cancelFn context.CancelFunc
```

Comment:
Variables should be declared close to the point of use. This isn't used until the deferred cancellation function, so let's declare it in the deferred callback itself rather than out here.